HospitalSystem.Core/People/Doctor/DoctorService.cs

```diff
@@ -56,4 +56,20 @@ public Doctor GetById(ObjectId id)
     {
         return _doctorRepo.GetById(id);
     }
+
+    
+    public int CompareCheckupsByDoctorsName(Checkup checkup1, Checkup checkup2)
+    {
+        string name1 = GetById((ObjectId)checkup1.Doctor.Id).FirstName;
+        string name2 = GetById((ObjectId)checkup2.Doctor.Id).FirstName;
+        return String.Compare(name1, name2);
+    }
+
+    public int CompareCheckupsByDoctorsSpecialty(Checkup checkup1, Checkup checkup2)
+    {
+        string specialty1 = GetById((ObjectId)checkup1.Doctor.Id).Specialty.ToString();
+        string specialty2 = GetById((ObjectId)checkup2.Doctor.Id).Specialty.ToString();
+        return String.Compare(specialty1, specialty2);
+    }
```

Comment:
Might not belong in doctor service. Why not just pass two doctor ids? maybe

Reply:
used to be in appointment sevice, didnt belong there either. Tell patient to fix it!